packages/editor/src/components/entities-saved-states/test/use-is-dirty.js

```diff
@@ -0,0 +1,94 @@
+/**
+ * External dependencies
+ */
+import { act, renderHook } from '@testing-library/react';
+
+/**
+ * Internal dependencies
+ */
+import { useIsDirty } from '../hooks/use-is-dirty';
+
+jest.mock( '@wordpress/data', () => {
+	return {
+		useSelect: jest.fn().mockImplementation( ( fn ) => {
+			const select = () => {
+				return {
+					__experimentalGetDirtyEntityRecords: jest
+						.fn()
+						.mockReturnValue( [
+							{
+								kind: 'root',
+								name: 'site',
+								title: 'title',
+								property: 'property',
+							},
+							{
+								kind: 'postType',
+								name: 'post',
+								title: 'title',
+								property: 'property',
+							},
+						] ),
+					getEntityRecordEdits: jest.fn().mockReturnValue( {
+						title: 'My Site',
+					} ),
+				};
+			};
+			return fn( select );
+		} ),
+	};
+} );
+
+jest.mock( '@wordpress/core-data', () => {
+	return {
+		store: {
+			__experimentalGetDirtyEntityRecords: jest.fn(),
+			getEntityRecordEdits: jest.fn(),
+		},
+	};
+} );
+
+describe( 'useIsDirty', () => {
+	it( 'should calculate dirtyEntityRecords', () => {
+		const { result } = renderHook( () => useIsDirty() );
+		expect( result.current.dirtyEntityRecords ).toEqual( [
+			{
+				kind: 'postType',
+				name: 'post',
+				property: 'property',
+				title: 'title',
+			},
+			{ kind: 'root', name: 'site', property: 'title', title: 'Title' },
```

Comment:
Shouldn't this be "My Site"?

Reply:
No, it assigns `property`(not variable) for `property`!
https://github.com/WordPress/gutenberg/blob/trunk/packages/editor/src/components/entities-saved-states/hooks/use-is-dirty.js#L39

This is one of the reasons I added a unit test for this hook rather than catching it by E2E. It's a bit complicated logic that might be a cause of an unexpected bug in the future. I wanted to protect this hook at the bottom of the testing pyramid.